src/redfish/messages/messages.py

```diff
@@ -0,0 +1,182 @@
+#! /usr/bin/python
+# Copyright Notice:
+# Copyright 2019-2020 DMTF. All rights reserved.
+# License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/Redfish-Tacklebox/blob/main/LICENSE.md
+
+"""
+Messages Module
+
+File : messages.py
+
+Brief : This file contains the definitions and functionalities for interacting
+        with Messages for a given Redfish service
+"""
+import re
+
+class RedfishOperationFailedError( Exception ):
+    """
+    Raised when an operation has failed (HTTP Status >= 400)
+    """
+    pass
+
+class RedfishPasswordChangeRequiredError( Exception ):
+    """
+    Raised when password change required
+    """
+    def __str__(self):
+        return  "\n{}\nURL: {}\n".format( str(self.args[0]), str(self.args[1]) )
+
+def get_messages_detail( response ):
+    """
+    Builds messages detail dict in the payload
+
+    Args:
+        response: The response to parser
+
+    Returns:
+        The dict containing messages_detail
+        messages_detail["status"]: http status code
+        messages_detail["successful"]: response successful (http status code < 400)
+        messages_detail["code"]: redfish message response code field
+        messages_detail["@Message.ExtendedInfo"]: redfish message response code field
+    """
+
+    messages_detail = {}
+    messages_detail["status"] = response.status
+    messages_detail["successful"] = False
+    messages_detail["@Message.ExtendedInfo"] = []
+
+    if response.status >= 400:
+        messages_detail["successful"] = False
+    else:
+        messages_detail["successful"] = True
+
+    try:
+        message_body = response.dict
+        if not "@Message.ExtendedInfo" in message_body:
+            message_body = response.dict["error"]
+        check_message_field = True
+        if "@Message.ExtendedInfo" in message_body:
+            messages_detail["@Message.ExtendedInfo"] = message_body["@Message.ExtendedInfo"]
+            for index in range(len(messages_detail["@Message.ExtendedInfo"])):
```

Comment:
Thinking about this for loop a bit, I don't think we want to insert empty strings if fields are not found. An empty string would convey to a user that some information was returned by the service, but it's not very useful. Some of these strings are expected to have programmatic structure, and an empty string would break that pattern (such as `@odata.type` always in the form of `#<NAMESPACE>.<TYPE>`.

It's probably safer at this time to just copy the array over as-is and not add any blank properties.

Reply:
Perhaps at least keeping a blank MessageId is okay since it is mandatory per the spec, and the absence of a MessageId can break clients. However, all other properties are optional.